src/irc_bridge.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Provide the IRC bridge class to represent the matrix-appservice-app plugin for Synapse."""
+
+# disabling due the fact that collect status does many checks
+# pylint: disable=too-many-return-statements
```

Comment:
Perhaps group the checks into separate functions?

Reply:
Wouldn't that add overhead particular to this setup? The `return` conditions add a logger or a MaintenanceStatus before returning, otherwise I would group them here as well and return once.

Reply:
Let's leave this for now since I think this function will need to be reworked anyway